shipctl/x86_64/WindowsServer_2016/utility.ps1

```diff
@@ -630,85 +658,116 @@ function notify() {
   $default_slack_payload = '{"username":"$opt_username","attachments":[{"pretext":"$opt_pretext","text":"$opt_text","color":"$opt_color"}],"channel":"$opt_recipient","icon_url":"$opt_icon_url"}'
   $default_webhook_payload = '{"username":"$opt_username","pretext":"$opt_pretext","text":"$opt_text","color":"$opt_color","channel":"$opt_recipient","icon_url":"$opt_icon_url"}'
   $default_payload=""
+  if ($r_method -eq "irc") {
+    if (!$env:opt_recipient) {
+      $recipients_list = $(get_resource_version_key "$env:opt_resource" "recipients")
+    }
 
-  if ($r_mastername -eq "Slack" -or $r_mastername -eq "slackKey") {
-    $default_payload=$default_slack_payload
-    $recipients_list = $(get_resource_version_key "$env:opt_resource" "recipients")
-  } elseif ($r_mastername -eq "webhook" -or $r_mastername -eq "webhookV2") {
-    $r_authorization=$(get_integration_resource_field "$env:opt_resource" authorization)
-    if (!$r_authorization) {
-      $r_authorization = ""
+    if ($($recipients_list.count) -gt 0) {
+      foreach ($recipient in $recipients_list) {
+        $server,$channel = $recipient.split('#')
+        if (!$server) {
+          throw "Error: no server specified in recipient $recipient"
+        } elseif (!$channel) {
+          throw "Error: no channel specified in recipient $recipient"
+        }
+        $channel = "#$channel"
+        Write-Output "sending notification to $recipient"
+        _send_irc_notification -server $server -channel $channel -nick $env:opt_username -pass $env:opt_password -payload $env:opt_text
+      }
+    } else {
```

Comment:
There's a lot of duplication in these if/else cases. Wouldn't it be cleaner to normalize it and always operate on $recipients_list?

Reply:
hmm, yeah i'll try to clean this up